### ISSUE TITLE
Add a column start collapsed - #92

### ISSUE THREAD

**Issue by mui-org** [cdotyone/mochaui#32]
Reported by dennisva...@kpnplanet.nl, Feb 11, 2010

What does your patch do?
Start a column in collapsed mode in order to use it later.
Because columns cannot be created later on a first or second place (imagine
3 column design)

Usage:
generate left column:
new MUI.Column({
    id: 'sideColumn1',
    placement: 'left',
    width: 200,
    resizeLimit: [100, 300],
    isCollapsed: true
});

Open the column:
MUI.columnCollapse($('sideColumn1'));
Close the column:
MUI.columnExpand($('sideColumn1'));

What version of MochaUI is it for and what revision number in the SVN?
0.9.6
Revision 529

What files were changed?
Layout.js

What browsers have you tested it in?
mac, windows xp, ff3, safari, ie7, chrome

line 340 add:
        isCollapsed:   false,

line 474 add:
        if (this.options.isCollapsed){
            this.columnToggle();
        }

Next code based on idea from John Porra:
line 1409 add:
        /\* Function: columnCollapse 
         \* Collapse a Column  
         \* Will fire the onColapse event when finished.
         \* 
        */ 
        columnCollapse: function(columnEl){ 
            var instances = MUI.Columns.instances;
            var instance = instances.get(columnEl.id);
            if (instance.isCollapsed == false){ 
                columnEl.oldWidth = columnEl.getStyle('width').toInt(); 
                instance.resize.detach(); instance.handleEl.removeEvents('dblclick'); 
                instance.handleEl.addEvent('click', function(){ 
                    columnEl.columnExpand(); 
                }.bind(columnEl)); 
                instance.handleEl.setStyle('cursor', 'pointer').addClass('detached'); 
                columnEl.setStyle('width', 0); 
                instance.isCollapsed = true; 
                columnEl.addClass('collapsed'); 
                columnEl.removeClass('expanded'); 
                MUI.rWidth(); 
            } 
            columnEl.fireEvent('onCollapse'); 
            return true;
        }, 

```
    /* Function: columnExpand 
     * Expands a Column 
     *  Will fire the onExpand event when finished.
     * 
    */ 
    columnExpand : function(columnEl){ 
        var instances = MUI.Columns.instances;
        var instance = instances.get(columnEl.id);
        if (instance.isCollapsed == true){ 
            columnEl.setStyle('width', columnEl.oldWidth); 
            instance.isCollapsed = false; 
            columnEl.addClass('expanded'); 
            columnEl.removeClass('collapsed');
            instance.handleEl.removeEvents('click'); 
            instance.handleEl.addEvent('dblclick', function(){ 
                columnEl.columnCollapse(); 
            }.bind(columnEl)); 
            instance.resize.attach(); 
            instance.handleEl.setStyle('cursor', Browser.Engine.webkit ? 'col-
```

resize' : 'e-resize').addClass('attached');
                MUI.rWidth(); 
            } 
            columnEl.fireEvent('onExpand');
            return true;
        },

Start reviewing: /branches/
Comment 2 by dennisva...@kpnplanet.nl, Feb 11, 2010

2 changes:
function columnCollapse
change line:
   columnEl.oldWidth = columnEl.getStyle('width').toInt(); 
to:
   instance.oldWidth = columnEl.getStyle('width').toInt(); 

and in 
function columnExpand
change:
  columnEl.setStyle('width', columnEl.oldWidth);
to:
  columnEl.setStyle('width', instance.oldWidth); 

sorry for inconvenience

**Comment by mui-org** [comment]
done with minor changes
